state/machine.go

```diff
@@ -1419,12 +1439,13 @@ func (m *Machine) DesiredSpaces() (set.Strings, error) {
 // instance will be able to act for that machine.
 func (m *Machine) SetProvisioned(
 	id instance.Id,
+	displayName instance.DisplayName,
 	nonce string,
 	characteristics *instance.HardwareCharacteristics,
 ) (err error) {
 	defer errors.DeferredAnnotatef(&err, "cannot set instance data for machine %q", m)
 
-	if id == "" || nonce == "" {
+	if id == instance.UnknownId || nonce == "" {
```

Comment:
Kind of unnecessary, but I added it in as the constant has been defined within `core/instance` and using it means IDEs can make use of it.